mParticleCore.brs

```diff
@@ -369,6 +404,92 @@ function mParticleConstants() as object
             product.cc = couponCode
         end function
     }
+    
+    '
+    ' Media APIs
+    '
+    
+    MediaSession = {
+        build : function(contentID as string, title as string, contentType as string, streamType as string, duration=0 as integer)
+            session = {}
+            session.contentID = contentID
+            session.title = title
+            session.duration = duration
+            session.contentType = contentType
+            session.streamType = streamType
+            session.mediaSessionID = CreateObject("roDeviceInfo").GetRandomUUID()
+            session.currentPlayheadPosition = 0
+            return session
+        end function,
+        setDuration : function(session as object, duration as integer)
+            session.duration = duration
+        end function,
+        setCurrentPlayheadPosition : function(session as object, playheadPosition as integer)
+            session.currentPlayheadPosition = playheadPosition
+        end function,
+        setAdContent : function(session as object, adContent as object)
+            session.adContent = adContent
+        end function,
+        setAdBreak : function(session as object, adBreak as object)
+            session.adBreak = adBreak
+        end function,
+        setSegment : function(session as object, segment as object)
+            session.segment = segment
+        end function
+    }
+    
+    AdContent = {
+        build : function(id as string, title as string)
+            ad = {}
+            ad.id = id
+            ad.title = title
+            return ad
+        end function,
+        setDuration : function(ad as object, duration as integer)
+            ad.duration = duration
+        end function,
+        setAdvertiser : function(ad as object, advertiser as string)
+            ad.advertiser = advertiser
+        end function,
+        setCampaign : function(ad as object, campaign as string)
+            ad.campaign = campaign
+        end function,
+        setCreative : function(ad as object, creative as string)
+            ad.creative = creative
+        end function,
+        setPlacement : function(ad as object, placement as string)
+            ad.placement = placement
+        end function,
+        setIndex : function(ad as object, index as integer)
```

Comment:
I think we refer to `index` as `position` in the docs.

https://docs.mparticle.com/developers/sdk/web/media#logging-advertising

Reply:
Will fix. I think I saw that we use index for segment and just felt it was clearer (segment_index is a key for segment on the same page)